mono/metadata/class.c

```diff
@@ -515,6 +521,18 @@ mono_type_get_name_recurse (MonoType *type, GString *str, gboolean is_recursed,
 			MonoTypeNameFormat nested_format;
 			int i;
 
+			// https://bugzilla.xamarin.com/show_bug.cgi?id=57938
+			// Generic instances containing generic parameters from themselves
+			//  should have a null fullname unless they are generic type definitions
+			if (!mono_class_is_gtd (klass)) {
```

Comment:
This is capturing a long more than just the GTD.

What about open generic instances with arguments from other types?